### PR TITLE
Blog onboarding: Update celebration screen to support design-first flow

### PR DESCRIPTION
**Pull request by paulopmt1**
Related to 2522-gh-Automattic/dotcom-forge

## Proposed Changes

- Make sure we don't have the gray line at the top
- Change the subtitle according to the next action (Connect to social or writing the first post)
- Preview has enough size to show the site
- Ensure selected domain appears (if user selected one)
- Define what to do on preview in case user didn't write a first post. Started a [thread about it here](p1684879946959299/1684879942.447799-slack-CKZHG0QCR).
- If user published the first post show Connect to social button, otherwise show Write your first post CTA
- Create a test for it

## Testing Instructions

- Load [this Jetpack PR](https://github.com/Automattic/jetpack/pull/30871) to your WPCOM Sandbox, but since the Jetpack PR is now merged, if you can't load it with the command above, you can use this diff instead D111566-code
- Go to /setup/design-first on an account with 0 existing sites.
- Launch a free site with no custom domain
- Run unit tests with `yarn test-client client/landing/stepper/declarative-flow/internals/steps-repository/celebration-step/test/index.test.tsx`
- Ensure `start-writing` celebration screen also works as expected, since we're reusing that code.
- Ensure we have `Now it’s time to start posting.` as subtitle and `Write your first post` CTA when user didn't write a post.
- Create and publish a post for the current site in a new tab, refresh the celebration screen and you should see `Now it’s time to connect your social accounts.` as subtitle and `Connect to social`. 

 

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/blog-onboarding-translations.js
Comment (paulopmt1) [inline client/landing/stepper/declarative-flow/blog-onboarding-translations.js:1]: We don't need this file anymore, so I take advantage we're testing our flow to remove it.

File: client/landing/stepper/declarative-flow/internals/steps-repository/celebration-step/index.tsx
Comment (paulopmt1) [inline client/landing/stepper/declarative-flow/internals/steps-repository/celebration-step/index.tsx:1]: I renamed the file. This isn't entirely new. Most of the change is between lines 35-58

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8008538

Hi @paulopmt1, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Now it’s time to start posting.`

Thank you in advance!

**Review by DustyReagan — APPROVED** [review]
Looks good to me!

I noticed one issue, but I'm not sure it's actually to do with the celebration-step, so it might be out of scope for this PR. But, if you click "Go to dashboard" on the design-first celebration step, you get redirected to the hero flow. 🤔 


https://github.com/Automattic/wp-calypso/assets/140841/0ecf4b19-455d-4b4b-8058-14f0b481dc57

**Comment by candy02058912** [comment]
> I noticed one issue, but I'm not sure it's actually to do with the celebration-step, so it might be out of scope for this PR. But, if you click "Go to dashboard" on the design-first celebration step, you get redirected to the hero flow. 🤔

@DustyReagan I have noticed that the same issue seems to be happening with https://github.com/Automattic/wp-calypso/pull/77222, which could suggest that it's not necessarily related to the celebration-step.

**Review by candy02058912 — APPROVED** [review]
The celebration step is working as intended! LGTM
design-first
design-first
start-writing
start-writing

**Comment by paulopmt1** [comment]
> @DustyReagan I have noticed that the same issue seems to be happening with #77222, which could suggest that it's not necessarily related to the celebration-step.

Yeah, I was able to reproduce that, and as @candy02058912 pointed it's a broader issue. It seems related to [the fact we remove our intent here](https://github.com/Automattic/wp-calypso/blob/update/celebration-screen/client/landing/stepper/declarative-flow/design-first.ts#L107) so [areLaunchpadTasksCompleted will always return false](https://github.com/Automattic/wp-calypso/blob/update/celebration-screen/client/my-sites/customer-home/controller.jsx#L57).

I'll open a follow-up task to investigate that better.

**Comment by paulopmt1** [comment]
Once https://github.com/Automattic/wp-calypso/pull/77222 gets merged, we'll merge this one.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8008538).

**Pull request merged**